### ISSUE TITLE
How did you learn?

### ISSUE THREAD

**Issue by whereeaglesfly** [mdo/feedback#140]
Hi man. How did you learn HTML & CSS ? Also JS? Are there any book, resource  which you followed? Do you recommend MDN Network? Share your some knowledge.

**Comment by mdo** [comment]
Taught myself years ago by copy-pasting snippets at first and then dissecting other sites to recreate them. Never read any books about it. Years ago I followed folks and sites that nowadays don't even seem all too relevant haha given the stuff they write about. I read the specs in a lot of instances, too.

Today, MDN is huge for me. It's the best resource to learn about what elements/properties are available, how to use them, etc. Super cool site.

Sorry I don't have much more to share there :).

**Comment by sukrosono** [comment]
So you just really flowing with whatever question in your mind?
give us some tips dude?

**Comment by mdo** [comment]
> give us some tips dude?

The only advice I can give is from what I've learned and how I've learned it. I've been doing this for 16+ years now. I got to where I am by reading the specs, reading blogs, dissecting examples and demos I loved, and trying to make new things.

I illegally downloaded software, made random things for absolutely no one and posted it to DeviantArt for free, I tried new things (used to make themes for Windows XP), and I wrote code every day for as long as I could.

If you want to learn HTML and CSS, practice writing it all the time. That's all I've done—I've just been doing it for a super long time :).
